tests/Agent/UnitTests/CompositeTests/CompositeTestAgent.cs

```diff
@@ -261,6 +311,9 @@ public void Dispose()
             var transaction = _primaryTransactionContextStorage.GetData();
             transaction?.Finish();
 
+            // reset the bootstrap configuration in case this was a serverless mode test
+            ConfigurationLoader.UseBootstrapConfigurationForTesting(BootstrapConfiguration.GetDefault());
```

Comment:
Instead of restoring the result of GetDefault, perhaps the composite test agent should capture the original value before swapping it out, and then restore that original value here?

Reply:
Done